pkg/kubectl/cmd/annotate.go

```diff
@@ -114,6 +114,7 @@ func NewCmdAnnotate(f *cmdutil.Factory, out io.Writer) *cobra.Command {
 	cmd.Flags().StringVar(&options.resourceVersion, "resource-version", "", "If non-empty, the annotation update will only succeed if this is the current resource-version for the object. Only valid when specifying a single resource.")
 	usage := "Filename, directory, or URL to a file identifying the resource to update the annotation"
 	kubectl.AddJsonFilenameFlag(cmd, &options.filenames, usage)
+	cmdutil.AddRecursiveFlags(cmd)
```

Comment:
It would be better to have a "recursive" field in the options struct, bind the flag to that field (possibly with a helper to make the flag name centralized and consistent), and check the options struct field below